### ISSUE TITLE
Supertab doesn't work in command-line window

### ISSUE THREAD

**Issue by mimooh** [ervandew/supertab#199]
After going to command-line window via q: the <tab> completion is gone. Any ideas?

**Comment by ervandew** [comment]
Hmm, seems to work fine for me. Do you have `paste` (`:set paste?`) enabled in your command window?

**Comment by mimooh** [comment]
Hmm, paste returns 'nopaste', which looks as it should be. If it works
for you, it may be that some other plugins of mine interfere with it.
But command-line window is the only place where I have issues with
supertab. 

Ctrl-X n autocompletes, just supertab doesn't.

-- 
Karol Kreński
tel. 517-302-055

**Comment by mimooh** [comment]
I figured it out somewhat. Fist this is that we may use different compilations of vim:

"When Vim was compiled without the +cmdline_compl feature only file names,
directories and help items can be completed.  The number of help item matches
is limited (currently to 300) to avoid a long delay when there are very many
matches."

I finally noticed I have this behaviour:
In a normal window:
i  <Tab>         <Plug>SuperTabForward

In the command line window:
i  <Tab>        @<C-X><C-V>
i  <Tab>         <Plug>SuperTabForward


Then I found the exactly same, unresolved problem that I face:
https://groups.google.com/forum/#!topic/vim_use/EX93BOy4XZM/discussion

QUOTE
I am trying to disable the default mapping for the tab key to @a<c-x><c-r> in command window. according to :h command-line-window, adding the following two lines in .vimrc should do the job but it doesn't to my test.
au CmdwinEnter [:>] iunmap <Tab>
au CmdwinEnter [:>] nunmap <Tab>
below is the min repo vimrc I observed this behavior.
set nocompatible
au CmdwinEnter [:>] iunmap <Tab>
au CmdwinEnter [:>] nunmap <Tab>

once in command window (either by : then ctrl-f, or q: in normal mode), :map<tab> still shows "n <Tab> @a<C-X><C-V>", which conflicts/overwrites my other mapping for the tab key.

set wildchar=<non-tab> seems to avoid the default mapping, but I do want to use tab for auto-completion in vim. I also tried tightening the scope of non-tab wildchar with
au CmdwinEnter [:>] set wildchar=<C-E>
au CmdwinLeave [:>] set wildchar=<Tab>
but this doesn't work either.

meanwhile, with vim help recommended setting, I always got this error on entering command window:
Error detected while processing CmdwinEnter Auto commands for "[:>]":
E31: No such mapping
Press ENTER or type command to continue

it feels like by the time CmdwinEnter is triggered, the default tab mapping is not injected yet, and somehow it comes in at a later phase, but still before my "set wildchar=<C-E>" takes effect, which is really confusing.

how can I disable them permanently? I am on vim 8.0.1420. thanks.
END OF QUOTE
